packages/backend/src/services/ValidationService/ValidationService.ts

```diff
@@ -244,7 +245,20 @@ export class ValidationService {
                 CreateChartValidation,
                 'error' | 'errorType' | 'fieldName'
             >) => {
-                if (!fieldIds?.includes(fieldId)) {
+                const { tableName } = chart;
+                const joinedTables =
+                    explores
+                        .find((e) => e.name === tableName)
+                        ?.joinedTables.map((jt) => jt.table) || [];
```

Comment:
I think there will be issues with join alias.  Could we add unit tests for this?

I was looking at the code and expected the `fieldIds` argument to only have valid values. 
And having all the `explores` as an argument defeats the purpose of the `explorerFields: CompiledField[],` argument. 

Should we precalculate all the valid field ids per explore and have the argument `exploreFields: Record<string, { dimensionIds: string[]; metricIds: string[]}>` ?
This way we only do it once per explore while now we do it multiple times per chart and field.

Reply:
Added tests and refactored code